src/mesh.cpp

```diff
@@ -476,7 +476,7 @@ void RegularMesh::bins_crossed(const Particle* p, std::vector<int>& bins,
       // The track exits this cell.  Determine the distance to the closest mesh
       // surface.
       xt::xtensor<double, 1> d = xt::zeros<double>({n});
-      for (int j = 0; j < n; ++j) {
+      for (j = 0; j < n; ++j) {
```

Comment:
I don't like using the same `j` for two different purposes here (as a loop index and then as an argmin index).

Additionally, I don't like how `j` is declared but never used if `ijk0 == ijk1`.  Until you notice the `break`, it looks like there is a code path for `j` to be used un-initialized.  I think we can clean up that logic (and fix outdated comments) with something like
```C++
    if (ijk0 == ijk1) {
      // The track ends in this cell.  Compute the distance to the particle's
      // end location, assign the next tally bin & weight, then finish looping.
      auto distance = (r1 - r0).norm();
      lengths.push_back(distance / total_distance);
      bins.push_back(get_bin_from_indices(ijk0.data()));
      break;
    }

    // The track exits this cell.  Determine the distance to the closest mesh
    // surface.
    xt::xtensor<double, 1> d = xt::zeros<double>({n});
    for (k = 0; k < n; ++k) {
      if (std::fabs(u[k]) < FP_PRECISION) {
        d(k) = INFTY;
      } else if (u[k] > 0) {
        double xyz_cross = lower_left_[k] + ijk0(k) * width_[k];
        d(k) = (xyz_cross - r0[k]) / u[k];
      } else {
        double xyz_cross = lower_left_[k] + (ijk0(k) - 1) * width_[k];
        d(k) = (xyz_cross - r0[k]) / u[k];
      }
    }

    // Assign the next tally bin & weight.
    auto j = xt::argmin(d)(0);
    auto distance = d(j);
    lengths.push_back(distance / total_distance);
    bins.push_back(get_bin_from_indices(ijk0.data()));

    // Translate the starting coordintes by the distance to the oncoming mesh
    // surface.
    r0 += distance * u;

    // Increment the indices into the next mesh cell.
    if (u[j] > 0.0) {
      ++ijk0(j);
    } else {
      --ijk0(j);
    }

    // If the next indices are invalid, then the track has left the mesh and
    // we are done.
    if (xt::any(ijk0 < 1) || xt::any(ijk0 > shape_)) break;
```